e2e-tests/cypress/tests/integration/channels/channel/channel_header_modal_spec.js

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channels @channel @channel_settings
+
+describe('Channel Settings - Channel Header', () => {
+    let testTeam;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, user}) => {
+            testTeam = team;
+            cy.apiLogin(user);
+
+            // # Visit town-square channel
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+        });
+    });
+
+    it('Create channel, modify header, and verify preview button functionality', () => {
+        // # Create a new channel
+        cy.apiCreateChannel(testTeam.id, 'new-channel', 'New Channel').then(({channel}) => {
+            // # Visit the newly created channel
+            cy.visit(`/${testTeam.name}/channels/${channel.name}`);
+
+            // # Click on the channel name in the channel header to open the channel menu options
+            cy.get('#channelHeaderDropdownIcon').click();
+
+            // # Select the "Edit Channel Header" option from the dropdown
+            cy.findByText('Edit Channel Header').click();
+
+            // # Type something in the header edit box
+            cy.get('#edit_textbox').clear().type('This is the new header content');
```

Comment:
instead of `#edit_textbox` can we have by placeholder or similar?

Reply:
I will replace this to get by aria-label.